src/woo_search/search_index/index.py

```diff
@@ -72,9 +86,14 @@ class Index:
 
 class Publication(ES_Document):
     uuid: M[str] = mapped_field(Text(required=True))
-    publisher: M[PublisherType] = mapped_field(Object(Publisher, required=True))
-    informatie_categorieen: M[List[InformatieCategorieType]] = mapped_field(
-        Nested(InformatieCategorie, required=True)
+    publisher: M[NestedPublisherType] = mapped_field(
+        Object(NestedPublisher, required=True)
+    )
+    informatie_categorieen: M[List[NestedInformationCategoryType]] = mapped_field(
```

Comment:
```suggestion
    informatie_categorieen: M[list[NestedInformationCategoryType]] = mapped_field(
```
and please fix this everywhere, I hadn't even realized that `typing.List` is used but this is Python 3.12, we can use the builtins.